src/main/java/io/confluent/connect/jdbc/dialect/Db2DatabaseDialect.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.connect.jdbc.dialect;
+
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.connect.data.Date;
+import org.apache.kafka.connect.data.Decimal;
+import org.apache.kafka.connect.data.Time;
+import org.apache.kafka.connect.data.Timestamp;
+
+import java.util.Collection;
+
+import io.confluent.connect.jdbc.dialect.DatabaseDialectProvider.SubprotocolBasedProvider;
+import io.confluent.connect.jdbc.sink.metadata.SinkRecordField;
+import io.confluent.connect.jdbc.util.ColumnId;
+import io.confluent.connect.jdbc.util.ExpressionBuilder;
+import io.confluent.connect.jdbc.util.ExpressionBuilder.Transform;
+import io.confluent.connect.jdbc.util.IdentifierRules;
+import io.confluent.connect.jdbc.util.TableId;
+
+/**
+ * A {@link DatabaseDialect} for IBM DB2.
+ */
+public class Db2DatabaseDialect extends GenericDatabaseDialect {
+
+  /**
+   * The provider for {@link Db2DatabaseDialect}.
+   */
+  public static class Provider extends SubprotocolBasedProvider {
+    public Provider() {
+      super(Db2DatabaseDialect.class.getSimpleName(), "db2", "db2j", "ibmdb");
+    }
+
+    @Override
+    public DatabaseDialect create(AbstractConfig config) {
+      return new Db2DatabaseDialect(config);
+    }
+  }
```

Comment:
Here's an example of how a dialect typically defines a custom `DatabaseDialectProvider` implementation class. It extends the `SubprotocolBasedProvider` so that it perfectly matches any JDBC URL with the `db2`, `db2j`, or `imdb` subprotocols, and it provides a factory method for the actual `Db2DatabaseDialect` implementation class.

Once again, this class is never directly used or referenced anywhere. Instead, the fully-qualified name of this provider class is specified in the `META-INF/services/io.confluent.connect.jdbc.dialect.DatabaseDialectProvider` file